### PR TITLE
Hold plots as in matplotlib default

### PR DESCRIPTION
**Pull request by f-brinkmann**
### Which issue(s) are closed by this pull request?

Adresses the hold fcuntionality requested in issue #16  

### Changes proposed in this pull request:

- added hold functionality

### CODE REVIEW

**Review by sikersten — CHANGES_REQUESTED** [review]
Solves the hold bug for figures with one axis (time, freq...). Unfortunately, replotting into figures with 2 subplots (e.g. freq_phase) fails. Should this issue be solved separately?

**Comment by f-brinkmann** [comment]
> 
> 
> Solves the hold bug for figures with one axis (time, freq...). Unfortunately, replotting into figures with 2 subplots (e.g. freq_phase) fails. Should this issue be solved separately?

hm, it works for me. This should yield a 2x1 subplot with two lines in the lower plot. Does this work for you guys?
~~~
import pyfar.plot as pp
from pyfar import Signal

X = Signal(np.sin(np.linspace(0, 20*np.pi, 512)), 44100)
Y = Signal(np.random.random(512), 44100)

plt.subplot(2, 1, 1)
pp.line.time(X)
plt.subplot(2, 1, 2)
pp.line.time(Y)
pp.line.time(X)
~~~

For me it works with and without passing axes objects to `pp.line.time`

**Comment by sikersten** [comment]
> 
> 
> > Solves the hold bug for figures with one axis (time, freq...). Unfortunately, replotting into figures with 2 subplots (e.g. freq_phase) fails. Should this issue be solved separately?
> 
> hm, it works for me. This should yield a 2x1 subplot with two lines in the lower plot. Does this work for you guys?
> 
> ```
> import pyfar.plot as pp
> from pyfar import Signal
> 
> X = Signal(np.sin(np.linspace(0, 20*np.pi, 512)), 44100)
> Y = Signal(np.random.random(512), 44100)
> 
> plt.subplot(2, 1, 1)
> pp.line.time(X)
> plt.subplot(2, 1, 2)
> pp.line.time(Y)
> pp.line.time(X)
> ```
> 
> For me it works with and without passing axes objects to `pp.line.time`

Sorry, I think my comment was not clear. What I meant is:
```
import pyfar.plot as pp
from pyfar import Signal

X = Signal(np.sin(np.linspace(0, 20*np.pi, 512)), 44100)
Y = Signal(np.random.random(512), 44100)

ax = pp.line.freq_phase(X)
pp.line.freq_phase(Y)
```
This does not work for me, does it work for you?

File: pyfar/plot/line.py [hunk 1]
```diff
@@ -379,21 +380,26 @@ def summary(signal, ax=None, style='light', **kwargs):
...
+    >>> # generate a signal
+    >>> s = Signal([0, 0, 1, 0, 0, 0, 0, 0, 0, 0], 1e3)
+    >>>
+    >>> # two by two plot of time and frequency signals
+    >>> ppl.multi(s, [[ppl.time, ppl.time_dB], [ppl.freq, ppl.group_delay]])
 
     """
-
     if not isinstance(signal, Signal):
         raise TypeError('Input data has to be of type: Signal.')
```
Comment (sikersten) [inline pyfar/plot/line.py:399]: Would a check is plots are pyfar handles make sense here?

**Review by sikersten — COMMENTED** [review]
A more general remark regarding most of the plot functions: the y-limits are set according to the 
last data plotted into the figure. This is different to the standard matplotlib behavior. Do we want it like that?

**Comment by f-brinkmann** [comment]
> A more general remark regarding most of the plot functions: the y-limits are set according to the
> last data plotted into the figure. This is different to the standard matplotlib behavior. Do we want it like that?

It's working nicely now :)

**Review by sikersten — APPROVED** [review]

File: pyfar/plot/line.py [hunk 2]
```diff
@@ -351,23 +351,24 @@ def freq_group_delay(signal, log_prefix=20, log_reference=1, ax=None,
...
+                                     ax, **kwargs)
 
     plt.tight_layout()
     Interaction('LineXLog', ax[0], signal, style, **kwargs)
 
     return ax
 
 
-def summary(signal, ax=None, style='light', **kwargs):
+def multi(signal, plots, ax=None, style='light', **kwargs):
```
Comment (mberz) [inline pyfar/plot/line.py:362]: Multi is a very arbitrary function name. Could we come up with something more explicit?

File: pyfar/plot/line.py [hunk 3]
```diff
@@ -351,23 +351,24 @@ def freq_group_delay(signal, log_prefix=20, log_reference=1, ax=None,
...
     """
-    Plot the time domain, the time domain in dB, the magnitude spectrum,
-    the frequency domain, the phase and group delay.
+    Generate multiple plots of a Signal object.
```
Comment (mberz) [inline pyfar/plot/line.py:364]: The docstring is also not very informational ;)

File: pyfar/plot/line.py (hunk 2 above)
Comment (f-brinkmann) [reply pyfar/plot/line.py:362]: I agree, but do not have the best ideas. I came up with:
- lines
- multi_line

File: pyfar/plot/line.py (hunk 3 above)
Comment (f-brinkmann) [reply pyfar/plot/line.py:364]: How about `Generate custom subplot layouts of a Signal object.`

Still not the best, but the example at the end can also help to clarify.

File: pyfar/plot/line.py (hunk 2 above)
Comment (sikersten) [reply pyfar/plot/line.py:362]: Could this function be extended to non-line plots? What about a name with something like subplots?

**Pull request merged**